### PR TITLE
gh-105486: Change the `repr` of `ParamSpec` list of args in `GenericAlias`

### PR DESCRIPTION
**Pull request by sobolevn**
Now it looks like this:

```python
>>> type A[X, **Y] = None
>>> A[int, [str]]
A[int, [str]]
```

* Issue: gh-105486

### CODE REVIEW

File: Lib/test/test_type_aliases.py [hunk 1]
```diff
@@ -142,7 +142,16 @@ def test_subscripting(self):
 
     def test_repr(self):
```
Comment (JelleZijlstra) [inline Lib/test/test_type_aliases.py:143]: Could you also add some tests to `test_genericalias.py`? This file is primarily intended to test the `type` statement, not `types.GenericAlias`.

File: Objects/genericaliasobject.c [hunk 2]
```diff
@@ -121,6 +121,39 @@ ga_repr_item(_PyUnicodeWriter *writer, PyObject *p)
...
 
+static int
+ga_repr_items_list(_PyUnicodeWriter *writer, PyObject *p)
+{
+    assert(PyList_CheckExact(p));
+
+    Py_ssize_t len = PyList_GET_SIZE(p);
+
+    if (_PyUnicodeWriter_WriteASCIIString(writer, "[", 1) < 0) {
+        goto error;
```
Comment (JelleZijlstra) [inline Objects/genericaliasobject.c]: As the error label is just `return -1`, I don't think we need the goto. The `goto error` pattern is useful if there is a lot of items to DECREF in the error path.

**Review by JelleZijlstra — COMMENTED** [review]
Thanks!

File: Objects/genericaliasobject.c (hunk 2 above)
Comment (sobolevn) [reply Objects/genericaliasobject.c]: (please, don't share my secret with anybody, but: I love `goto` statements).

Refactored! Thanks for the review :)

File: Lib/test/test_type_aliases.py (hunk 1 above)
Comment (sobolevn) [reply Lib/test/test_type_aliases.py:143]: Is it possible to create a pure and sensible `types.GenericAlias` with `ParamSpec` without a `type` statement? I cannot find any good examples.

Comment (JelleZijlstra) [reply Lib/test/test_type_aliases.py:143]: Not sure you could count this as sensible:

```
>>> class X:
...     __class_getitem__ = classmethod(types.GenericAlias)
... 
>>> X[[int, str]]
__main__.X[[<class 'int'>, <class 'str'>]]
```

Comment (sobolevn) [reply Lib/test/test_type_aliases.py:143]: Sounds good! 👍

**Comment by sobolevn** [comment]
Is there anything left?

**Review by JelleZijlstra — APPROVED** [review]

File: Objects/genericaliasobject.c (hunk 2 above)
Comment (JelleZijlstra) [reply Objects/genericaliasobject.c]: https://xkcd.com/292/

**Comment by JelleZijlstra** [comment]
@AlexWaygood you added the 3.12 backport label, but this feels like a feature change that should only go into main. What do you think?

Also, this PR is linked to issue #105487, but I don't see how it is related. I think this is a good change and we should merge it, but was there some other issue this should be linked to?

**Comment by JelleZijlstra** [comment]
Ah it should be #105486, I'll update the PR.

**Comment by AlexWaygood** [comment]
> @AlexWaygood you added the 3.12 backport label, but this feels like a feature change that should only go into main. What do you think?

Hmm, I view the current repr as pretty inconsistent — a latent bug that was exposed by the PEP-695 implementation. So I would favour backporting here. Given that 3.12 has yet to be released, I also think it will be less disruptive to change the repr now rather than have the repr change between 3.12 and 3.13, which could break people's doctests etc.

**Comment by JelleZijlstra** [comment]
That's fair. I suppose you could technically run into this in 3.11, but PEP 695 makes it a lot easier. I'm fine with backporting.

**Pull request merged**